WordPress/gutenberg
# Merged data should consider origin to return early

Part of https://github.com/WordPress/gutenberg/issues/45171
Related https://github.com/WordPress/gutenberg/pull/45971

## What?

This makes the `WP_Theme_JSON_Resolver::get_merged_data` consider the origin the consumers wants to retrieve, preventing it from calculating data consumers don't need.

## Why?

This PR is a preparation for https://github.com/WordPress/gutenberg/pull/45372/ and other public high-level methods. The consumer should be responsible to know which origin needs to be queried, and the resolver should return data based on that.

## How?

- This makes the `get_merged_data` method use the `$origin` parameter and bail early.
- Adds some test to clarify behavior.
- Brings back a couple of filters that were missing.

## Testing Instructions

- Verify that automated test pass.
- Test [scaling presets work as expected](https://github.com/WordPress/gutenberg/pull/45969#issuecomment-1324292280).
